src/main/c/Objects/pyjmethod.c

```diff
@@ -230,24 +268,19 @@ static PyObject* pyjmethod_call(PyJMethodObject *self,
     /* if params includes pyjarray instance */
     int            foundArray       = 0;
 
-    if (keywords != NULL && PyDict_Size(keywords) > 0) {
-        PyErr_Format(PyExc_RuntimeError, "Keywords are not supported.");
-        return NULL;
-    }
     lenPyArgsGiven = PyTuple_Size(args);
 
     env = pyembed_get_env();
     lenJArgsExpected = PyJMethod_GetParameterCount(self, env);
     if (lenJArgsExpected == -1) {
         return NULL;
     }
+    if (self->isKwArgs) {
+        lenJArgsExpected -= 1;
```

Comment:
I don't understand the +1 and -1 of len of args expected throughout this review.  Can you explain?

Reply:
lenJArgsExpected is how many regular python args need to be converted to java args. The easiest way to get the current code working with kwarrgs was to -1 here so it completely ignored the kwargs java expects and then add the kwargs at the end, That does make these variable names less accurate although I am having trouble coming up with a better name.

Reply:
In my latest change I have restructured the code so that when there are kwargs I decrement lenJArgsNormal instead of lenJArgsExpected, This really just means we +/- 1 in different places but overall I think it makes more sense because the kwargs are now included in expected but not included in normal. That makes sense to me because they are processed differently than other args so they are abnormal and expected. Let me know if you think the new way is worse, or if you have other ideas on how to make this cleaner.